topo_processor/metadata/metadata_loaders/metadata_loader_repo.py

```diff
@@ -17,23 +16,21 @@ class MetadataLoaderRepository:
     def append(self, loader: MetadataLoader) -> None:
         self.loaders.append(loader)
 
-    async def load_metadata(self, item: Item) -> None:
+    async def load_metadata(self, asset: Asset) -> None:
         async with self.lock:
             for loader in self.loaders:
-                if loader.is_applicable(item):
+                if loader.is_applicable(asset):
                     start_time = time_in_ms()
                     try:
-                        await loader.load_metadata(item)
-                    except Exception as error_msg:
-                        item.is_valid = False
-                        item.error_msgs.append(str(error_msg))
-                        get_log().warning(
-                            f"Metadata Load Failed: {error_msg}", loader=loader.name, source_path=item.source_path
-                        )
+                        await loader.load_metadata(asset)
+                        if not asset.is_valid:
+                            break
```

Comment:
i think add asset.item.is_valid = False

Reply:
metadata loader imagery historic will return before an item is created so at this point `asset.item = None`.

But you raise a good point we will have to be consistent with this when processing future datasets :thinking: